controller/fileItem.ts

```diff
@@ -0,0 +1,59 @@
+import Dangle from 'model/dangle';
+import { App, CachedMetadata, getLinkpath, TFile } from 'obsidian';
+import { Config } from '../service/config';
+import { ICONS, NAVIGATION_TAB } from '../constants';
+import { Sortable } from './navigationItem';
+
+export default class DanglingFileItem extends Sortable {
+    constructor(app: App) {
+        super(ICONS.FILE_ICON, NAVIGATION_TAB.FILE, 'File', app);
+    }
+
+    getDanglingLinks(): Map<string, Dangle[]> {
```

Comment:
Hmm, this code is reproduced in all 3 controllers -- could you factor it out?

Reply:
Factored it out. There is still some custom logic in each controller as the grouping is handled differently per controller.